jobs/build-podman-os.Jenkinsfile

```
@@ -0,0 +1,360 @@
+    supermin_run("--cache", arch, "qcow2", shortcommit)
+    def artifacts_runvm = artifacts.drop(1) // drop qcow2
+    parallel artifacts_runvm.collectEntries{artifact -> [artifact["platform"], {
+        supermin_run("--snapshot", arch, "${artifact["suffix"]}", shortcommit)
+    }]}
+}
+
+def generate_diskvar_json(shortcommit, arch, artifacts, repo) {
+    artifacts.each { artifact ->
+        def jsonContent = """
+        {
+            "osname": "fedora-coreos",
+            "container-repo": "$repo",
+            "container-tag": "${arch}-${shortcommit}",
+            "image-type": "${artifact["platform"]}",
+            "container-imgref": "ostree-remote-registry:fedora:quay.io/podman/machine-images:5.0",
+            "metal-image-size": "3072",
+            "cloud-image-size": "10240"
+        }
+        """
+        def file_path="./diskvars-${arch}-${artifact["suffix"]}.json"
+        writeFile file: file_path, text: jsonContent
+    }
+    if (arch != "x86_64") {
+        shwrap("""cosa remote-session sync ./diskvars-${arch}-*.json  {:,}""")
+    }
+}
+
+
+def supermin_run(cache, arch, artifact, shortcommit) {
+    shwrap("""cosa supermin-run ${cache} \
+       /usr/lib/coreos-assembler/runvm-osbuild \
+       --config ./diskvars-${arch}-${artifact}.json \
+       --filepath podman-$arch-${shortcommit}.${artifact} \
+       --mpp /usr/lib/coreos-assembler/osbuild-manifests/coreos.osbuild.${arch}.mpp.yaml
+    """)
+}
+
+properties([
+    pipelineTriggers([
+        [$class: 'GenericTrigger',
+         genericVariables: [
+          [
+           key: 'PODMAN_MACHINE_GIT_REF',
+           value: '$.ref',
+           expressionType: 'JSONPath',
+           regexpFilter: 'refs/heads/', //Optional, defaults to empty string
+           defaultValue: ''  //Optional, defaults to empty string
+          ]
+         ],
+         causeString: 'Triggered on $ref',
+         token: 'build-podman-os',
+         tokenCredentialId: '',
+         printContributedVariables: true,
+         printPostContent: true,
+         silentResponse: false,
+         regexpFilterText: '$PODMAN_MACHINE_GIT_REF',
+         regexpFilterExpression: 'main'
+        ]
+    ]),
+    parameters([
+      string(name: 'ARCHES',
+             description: 'Space-separated list of target architectures',
+             defaultValue: "x86_64 aarch64",
+             trim: true),
+      string(name: 'PODMAN_MACHINE_GIT_URL',
+             description: 'Override the coreos-assembler git repo to use',
+             defaultValue: "https://github.com/baude/podman-machine-images.git",
+             trim: true),
+      string(name: 'PODMAN_MACHINE_GIT_REF',
+             description: 'Override the coreos-assembler git ref to use',
+             defaultValue: "main",
+             trim: true),
+      string(name: 'CONTAINER_REGISTRY_REPO',
+             description: 'Override the registry to push the container to',
+             defaultValue: "quay.io/podman/machine-images",
```

From talking with brent I think this is where he is pushing too:

```suggestion
             defaultValue: "quay.io/podman/machine-os",
```